src/hooks/core/chat/useChatUtilsHook.ts

```diff
@@ -43,17 +46,19 @@ function useChatUtilsHook(): UseChatUtilsHook {
 
   const [selectedChannelKey, setSelectedChannelKey] = useRecoilState(selectedChannelKeyTab);
 
-  const {localDb, refresh} = useLocalDatabaseHook();
+  const {localDb, refresh, refreshWithId} = useLocalDatabaseHook();
   const navigation = useNavigation();
   const [profile] = (React.useContext(Context) as unknown as any).profile;
   const {anonProfileId, signedProfileId} = useUserAuthHook();
+  const {queue} = useDatabaseQueueHook();
+
   const setChannelAsRead = async (channel: ChannelList) => {
     if (!localDb) return;
     channel.setRead(localDb).catch((e) => console.log('setChannelAsRead error', e));
 
     if (channel?.channelType?.includes('ANON')) {
       try {
-        await AnonymousMessageRepo.setChannelAsRead(channel?.id?.replace('_anon', ''));
+        AnonymousMessageRepo.setChannelAsRead(channel?.id?.replace('_anon', ''));
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The error handling in the `setChannelAsRead` function could be improved. Currently, if an error occurs while calling `AnonymousMessageRepo.setChannelAsRead`, it is logged but the execution continues. It would be better to handle this error more gracefully, perhaps by showing a user-friendly message or retrying the operation.
```diff
     if (channel?.channelType?.includes('ANON')) {
       try {
        AnonymousMessageRepo.setChannelAsRead(channel?.id?.replace('_anon', ''));
-       } catch (error) {
-         console.log('setAnonChannelAsRead error api', error);
+       } catch (e) {
+         SimpleToast.show('Failed to mark channel as read. Please try again.');
+         console.error('setAnonChannelAsRead error api', e);
       }
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->